### PR TITLE
[DATA-2065] Make data required in file upload

### PR DESCRIPTION
**Pull request by njooma**

### CODE REVIEW

**Review by benjirewis — APPROVED** [review]
@purplenicole730 probably has more context than I do, but seems good!

**Comment by njooma** [comment]
@purplenicole730 is on vacay, so pinging @stuqdog instead

File: src/viam/app/data_client.py [hunk 1]
```diff
@@ -659,25 +659,27 @@ async def streaming_data_capture_upload(
...
         request_metadata = StreamingDataCaptureUploadRequest(metadata=metadata)
+        stream: Stream[StreamingDataCaptureUploadRequest, StreamingDataCaptureUploadResponse]
         async with self._data_sync_client.StreamingDataCaptureUpload.open(metadata=self._metadata) as stream:
             await stream.send_message(request_metadata)
             await stream.send_message(StreamingDataCaptureUploadRequest(data=data), end=True)
-            response: StreamingDataCaptureUploadResponse = await stream.recv_message()
+            response = await stream.recv_message()
             if not response:
                 await stream.recv_trailing_metadata()  # causes us to throw appropriate gRPC error
+                raise TypeError("Response cannot be empty")
```
Comment (stuqdog) [inline src/viam/app/data_client.py:669]: (q) Is it possible for us to actually reach this point? `if not response` is the case when the response isn't just empty, but a `None`, which I think should only happen if something went wrong such that we'd have a non-OK gRPC status and so throw an error at the preceding line. I guess either way there's no harm in having another safety rail here!

**Review by stuqdog — APPROVED** [review]
one question, but looks good to me!

File: src/viam/app/data_client.py (hunk 1 above)
Comment (njooma) [reply src/viam/app/data_client.py:669]: *WE* know we can't get here, but they *typechecker* doesn't know that and it was complaining

Comment (stuqdog) [reply src/viam/app/data_client.py:669]: Ahh got it, thanks!

**Pull request merged**